tests/unit/testTypesService.unitTest.ts

```diff
@@ -1,263 +1,410 @@
+      try {
+        await testTypesService.getTestTypesList();
+      } catch (errorResponse) {
+        expect(errorResponse.statusCode).toEqual(404);
+        expect(errorResponse.body).toEqual("No resources match the search criteria.");
+      }
+    });
+
+    context("getTestTypesById", () => {
+      it("should return error 404", async () => {
+        const MockTestTypesDAO = jest.fn().mockImplementation(() => {
+          return {
+            getAll: () => {
+              return Promise.resolve({
+                Items: [],
+                Count: 0
+              });
+            }
+          };
         });
+
+        const testTypesService = new TestTypesService(new MockTestTypesDAO());
+
+        try {
+          await testTypesService.getTestTypesById("1", {
+            fields: ["testTypeClassification", "defaultTestCode", "linkedTestCode"],
+            vehicleType: "psv",
+            vehicleSize: "small",
+            vehicleConfiguration: "rigid"
+          });
+        } catch (errorResponse) {
+          expect(errorResponse.statusCode).toEqual(404);
+          expect(errorResponse.body).toEqual("No resources match the search criteria.");
+        }
+      });
+    });
+  });
+});
+
+describe("when database is off", () => {
+  it("should return error 500", () => {
+
+    const MockTestTypesDAO = jest.fn().mockImplementation(() => {
+      return {
+        getAll: () => {
+          return Promise.reject({});
+        }
+      };
+    });
+
+    const testTypesService = new TestTypesService(new MockTestTypesDAO());
+
+    return testTypesService.getTestTypesList()
+      .then(() => undefined)
+      .catch((errorResponse) => {
+        expect(errorResponse.statusCode).toEqual(500);
+        expect(errorResponse.body).toEqual("Internal Server Error");
+      });
+  });
+});
+
+describe("fieldInfilterExpressionMatchesTheOneInTestCode", () => {
+  context("when passing a testCode that contains an array on field forVehicleAxles and a filterExpression that has a value included in that array  ", () => {
+    it("should return true", () => {
+      const testCode = JSON.parse("{\n" +
+        "                \"forVehicleType\": \"hgv\",\n" +
+        "                \"forVehicleSize\": null,\n" +
+        "                \"forVehicleConfiguration\": null,\n" +
+        "                \"forVehicleAxles\": [\n" +
+        "                  4,\n" +
+        "                  5,\n" +
+        "                  6,\n" +
+        "                  7,\n" +
+        "                  8,\n" +
+        "                  9,\n" +
+        "                  10\n" +
+        "                ],\n" +
+        "                \"defaultTestCode\": \"qqv\",\n" +
+        "                \"linkedTestCode\": null\n" +
+        "              }");
+
+      const filterExpression = {
+        vehicleAxles: 5,
+      };
+      // tslint:disable-next-line:no-empty
+      const MockTestTypesDAO = jest.fn().mockImplementation(() => { });
```

Comment:
can just use () => { return; } and then you don't need the tslint override above